src/Umbraco.Core/DependencyInjection/LegacyStaticServiceProvider.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.ComponentModel;
+
+namespace Umbraco.Cms.Web.Common.DependencyInjection;
+
+/// <summary>
+///     Service locator for internal (umbraco cms) only purposes. Should only be used if no other ways exist.
+/// </summary>
+/// <remarks>
+///     It is created with only two goals in mind
+///     1) Continue to have the same extension methods on IPublishedContent and IPublishedElement as in V8. To make
+///     migration easier.
+///     2) To have a tool to avoid breaking changes in minor and patch versions. All methods using this should in theory be
+///     obsolete.
+///     Keep in mind, every time this is used, the code becomes basically untestable.
+/// </remarks>
+[Obsolete("Use Umbraco.Cms.Core.DependencyInjection.StaticServiceProvider directly instead")]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class StaticServiceProvider
+{
+    /// <summary>
+    ///     The service locator.
+    /// </summary>
+    [Obsolete("Use Umbraco.Cms.Core.DependencyInjection.StaticServiceProvider directly instead")]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static IServiceProvider Instance { get; set; } = Core.DependencyInjection.StaticServiceProvider.Instance;
```

Comment:
@callumbwhyte I haven't tested this, but couldn't there potentially be some issues here if the legacy class is referenced before Umbraco has populated the new property?

![image](https://user-images.githubusercontent.com/3634580/197965984-01a4bd82-86fe-425b-ac0a-6de83a902544.png)

We're talking very edge cases, but this might be slightly better (I think):

```csharp
public static IServiceProvider Instance {
    get => Core.DependencyInjection.StaticServiceProvider.Instance;
    set => Core.DependencyInjection.StaticServiceProvider.Instance = value;
}
```

Reply:
@abjerner I agree, this is cleaner and less error prone. Will apply!